openstates/me/people.py

```diff
@@ -31,75 +31,99 @@ def scrape(self, chamber=None):
-                leg_url = district.xpath('a[3]/@href')[0]
-                leg_info_second_active = district.xpath('a[3]/text()')[0]
-
-                member_name = leg_info_second_active
-                mem_info = district.xpath('a[3]/following-sibling::text()')
-                party = _party_map[mem_info[0][2]]
-                district = mem_info[0].split('-')[1][:-1]
-            else:
-                member_name = info_search.group('member_name')
-                party = _party_map[info_search.group('party')]
-                district_name = info_search.group('district_name')
+        mapping = {
+            'district': 0,
+            'last_name': 1,
+            'first_name': 2,
+            'mid_name': 3,
+            'jr_sr': 4,
+            'party': 5,
+            'legal_res': 6,
+            'address1': 7,
+            'address2': 8,
+            'town': 9,
+            'state': 10,
+            'zip_code': 11,
+            'zip_plus4': 12,
+            'home_phone': 13,
+            'cell_phone': 14,
+            'business_phone': 15,
+            'other_phone': 16,
+            'fax': 17,
+            'email': 18,
+            'seat_no': 19,
+            'house_leadership': 20,
+            'committees': 21
+        }
 
-            # Get the photo url.
-            html = self.get(leg_url).text
-            doc = lxml.html.fromstring(html)
-            doc.make_links_absolute(leg_url)
-            (photo_url,) = doc.xpath('//img[contains(@src, ".jpg")]/@src')
+        url = (
+            'http://legislature.maine.gov/house/128house.xlsx'
+        )
+        fn, result = self.urlretrieve(url)
 
-            # Add contact information from personal page
-            office_address = re.search(
-                r'<B>Address:  </B>(.+?)\n?</?P>', html, re.IGNORECASE).group(1)
+        wb = xlrd.open_workbook(fn)
+        sh = wb.sheet_by_index(0)
 
-            office_email = doc.xpath(
-                '//a[starts-with(@href, "mailto:")]/text()')
-            business_phone = re.search(
-                r'<B>Business Telephone:  </B>(.+?)</?P>', html, re.IGNORECASE)
-            home_phone = re.search(
-                r'<B>Home Telephone:  </B>(.+?)</?P>', html, re.IGNORECASE)
-            cell_phone = re.search(
-                r'<B>Cell Telephone:  </B>(.+?)</?P>', html, re.IGNORECASE)
+        for rownum in range(1, sh.nrows):
+            # get fields out of mapping
+            d = {}
+            for field, col_num in mapping.items():
+                try:
+                    d[field] = str(sh.cell(rownum, col_num).value).strip()
+                except UnicodeEncodeError:
+                    # str typecasting will raise error for non-ascii characters.
+                    d[field] = sh.cell(rownum, col_num).value.encode('utf-8').strip()
+                except IndexError:
+                    # This col_num doesn't exist in the sheet.
+                    pass
+
+            district_number = d['district']
```

Comment:
If a member doesn't have a district (eg, the final two rows in the spreadsheet), they should be skipped.